Calibration/HcalAlCaRecoProducers/plugins/AlCaIsoTracksFilter.cc

```diff
@@ -351,11 +359,17 @@ bool AlCaIsoTracksFilter::filter(edm::Event& iEvent, edm::EventSetup const& iSet
 				   << " Chg Isolation " << hmaxNearP
 				   << ":" << eIsolation;
 	  if (t_p>pTrackMin_ && eMipDR<eEcalMax_ && hmaxNearP<eIsolation) {
-	    accept = true;
-	    break;
+	    if (t_p > pTrackLow_ && t_p < pTrackHigh_) ntrin++;
+	    else                                       ntrout++;
 	  }
 	}
       }
+      accept = (ntrout > 0);
+      if (!accept && ntrin > 0) {
+	++nRange_;
+	if      (preScale_ <= 1)         accept = true;
+	else if (nRange_%preScale_ == 1) accept = true;
+      }
     }
   }
   // Step 4:  Return the acceptance flag
```

Comment:
The idea is that if there are any tracks out of the range (20:40) and satisfying the minimum track threshold (20 GeV) we accept the event. If the event has only tracks which are within the range, then we select only a fraction of these events depending on the prescale factor. If prescale factor is 1, we accept all events with isolated track candidates above 20 GeV. Otherwise we accept all events with at least one isolated track candidate above 40 GeV and a fraction of events which have only isolated track candidate between 20 and 40 GeV.